### ISSUE TITLE
Large oak trees

### ISSUE THREAD

**Issue by VashaTS** [enjikaka/WoodCutter#3]
Sometimes it leaves part of large oak trees: http://i.imgur.com/XXmK4Gw.jpg

**Comment by enjikaka** [comment]
Thanks for letting me now! I'll look into it! 🍃

lör 5 dec 2015 18:05 VashaTS notifications@github.com skrev:

**Comment by RoyCurtis** [comment]
Whilst working on a 1.9 pull request, I tried testing and resolving this issue.

I believe the problem is that if there are multiple large oak trees very close to each other, chopping one down may remove enough of the other tree's logs to confuse the algorithm.

I've tried a few things (e.g. increasing the search radius, max offset from origin) and they did not seem to fix this issue.

I think the only way to fix this will be to rewrite the search algorithm, so that it:
- Also searches for leaf blocks, and considers them part of a tree to then look around for log blocks
- Keeps a record of blocks already searched; else, searching for leaf blocks causes an infinite loop

Just some ideas on this whilst testing today

**Comment by enjikaka** [comment]
I agree, the search algorithm needs to be refined. I might give it a shot
this weekend, or maybe you want to squash this bug? 🐞

Den mån 11 apr. 2016 16:49Roy Curtis notifications@github.com skrev:

**Comment by RoyCurtis** [comment]
I'd like to leave it up to you if you find the time :) Else I can try in a week or so
